test/model/store_test.dart

```diff
@@ -29,6 +31,36 @@ import 'test_store.dart';
 void main() {
   TestZulipBinding.ensureInitialized();
 
+  group('GlobalStore.updateGlobalSettings', () {
+    test('smoke', () async {
+      final globalStore = eg.globalStore();
+      check(globalStore).globalSettings.themeSetting.equals(null);
+      check(globalStore).globalSettings.browserPreference.equals(null);
+
+      final result = await globalStore.updateGlobalSettings(
+        GlobalSettingsCompanion(
+          themeSetting: Value(ThemeSetting.dark),
+          browserPreference: Value(BrowserPreference.external)));
+      check(globalStore).globalSettings.themeSetting.equals(ThemeSetting.dark);
+      check(globalStore).globalSettings.browserPreference.equals(
+        BrowserPreference.external);
```

Comment:
This doesn't need to test each setting as we add it; at that point we'd just be testing the generated code from Drift.